### DISCUSSION TITLE
Updating the container apps automatically

### DISCUSSION THREAD

**Discussion by kevdogg** [tteck/Proxmox#3607] [General]
Hey I using the traefik container and I just learned about the update command run inside the lxc container -- pretty slick.  Soo my question is how to actually automate this process.  Right now when I type update there is a box with a choice which you can choose y/n.  Can a choice be passed on the command line like update -y or update y?  Just wondering how to do this? Thanks.

**Comment by Mellowlynx** [comment]
Check here: [https://tteck.github.io/Proxmox/#proxmox-ve-cron-lxc-updater](https://tteck.github.io/Proxmox/#proxmox-ve-cron-lxc-updater) to set up a cron job.
Not recommended when you run semi-production or have LXC you depend on.

**Reply by kevdogg** [reply]
I believe this script just does things like apt update/upgrade etc correct?  The traefik LXC updater -- run within the LXC container checks the current traefik version vs a list of the currrent release, downloads the tar.gz file and then installs it.  I like the updater script a lot but would like to bypass the popup screen if possible so script could be run via ansible or such method.

**Reply by Mellowlynx** [reply]
You are correct, I misunderstood the question then.

Tteck already posted here somewhere, that he will not make a cronjob for the apps, every app is different, and something can break.
So better to do it manually after making a backup etc.

You are free to use the update script from an app to see how it updated and make your own update script and cron it.
But all at your own risk.
I get your point, but some services might be critical, so you need to plan the update before installing it, or better test first before deploying to 'production'.

As a side note:
I have only two apps (of the many) I update with Ansible, (pihole for example) the rest I update by hand when needed.
But this is a pretty save update and will not break my full lab.

**Reply by kevdogg** [reply]
Yea I get the breakage part, however this container a reverse proxy.  I suppose the download could fail however with this upgrade script which basically copies the new version of the traefik binary to /usr/bin/traefik it's super low I believe on the breakage level.  
With about 20 VMs spread across proxmox, xcp-ng and true-nas, at two locations, and random LXC containers, and the multiple docker stacks (even with use of watchtower), the once a week update time is ballooning and honestly I'm forgetting to do somethings.  I probably need to make more thorough notes however some form of automation would be very helpful as well to do simple things as this traefik LXC update.

**Comment by towerhand** [comment]
That would be nice, but I don't think it is possible with the current scripts. Therefore, I have been downloading and modifying them to run them weekly with a cron job.
